parser/remote/calcite/src/main/java/org/sqlflow/CalciteParserServer.java

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package org.sqlflow.parser.remote.calcite;
+
+import io.grpc.stub.StreamObserver;
+import java.io.IOException;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.sqlflow.parser.remote.ParserGrpc;
+import org.sqlflow.parser.remote.ParserProto;
+import org.sqlflow.parser.remote.ParserServer;
+
+public class CalciteParserServer extends ParserServer {
+
+  public static void main(String[] args) throws IOException, InterruptedException {
+    final CalciteParserServer s = new CalciteParserServer();
+    s.start(new CalciteParserImpl(), parsePort(args, 50053));
+    s.blockUntilShutdown();
+  }
+
+  static class CalciteParserImpl extends ParserGrpc.ParserImplBase {
```

Comment:
Any reason to use inner class here?

Reply:
No idea. Will move it to an outer level in the next PR.